doc/tutorials/inter-vm_communication.rst

```diff
@@ -0,0 +1,490 @@
+       :align: center
+       :name: Inter-VM ivshmem data transfer state machine
+
+       Inter-VM ivshmem data transfer state machine
+
+  :numref:`Inter-VM ivshmem handshake communication` shows the handshake communication between two machines:
+
+    .. figure:: images/Inter-VM_handshake_communication_two_machine.png
+       :align: center
+       :name: Inter-VM ivshmem handshake communication
+
+       Inter-VM ivshmem handshake communication
+
+
+**The reference Sender and Receiver Sample Code based Doorbell mode:**
+
+     .. code-block:: none
+        :emphasize-lines: 11,14,18,20,24,40
+
+        struct pay_load_header
+        {
+            uint64_t	p0_status;
+            uint64_t	p1_status;
+        };
+
+        void ivsh_test_sender(struct ivsh_dev_context *p_ivsh_dev_ctx, struct ivsh_ctrl_context *p_ivsh_ctrl_ctx)
+        {
+            struct ivsh_test_tx_context tx_ctx;
+            volatile struct pay_load_header *p_hdr;
+            /* Initialize the sender related data */
+            ivsh_test_tx_init(&tx_ctx, p_ivsh_dev_ctx, p_ivsh_ctrl_ctx);
+            p_hdr = tx_ctx.p_hdr;
+            /*Set P0 status to RESET*/
+            set_p0_status(p_hdr, SHMEM_STATUS_RESET);
+            while (!is_p1_reset(p_hdr))
+                usleep(10000);
+            /*Prepare the data to be sent */
+            ivsh_test_tx_pre_send(&tx_ctx, i);
+            /*Set P0 status to INIT*/
+            set_p0_status(p_hdr, SHMEM_STATUS_INIT);
+            while (!is_p1_initialized(p_hdr)) {
+            }
+            /*Set P1 status to READY*/
+            set_p1_status(p_hdr, SHMEM_STATUS_READY);
+            usleep(2000);
+
+            ivsh_test_tx_send(&tx_ctx);
+            ivsh_trigger_doorbell(p_ivsh_dev_ctx, p_ivsh_ctrl_ctx->peer_id, IVSH_TEST_VECTOR_ID);
+            ivsh_wait_irq(p_ivsh_dev_ctx, IVSH_TEST_VECTOR_ID);
+
+            ivsh_test_tx_deinit(&tx_ctx);
+        }
+
+        void ivsh_test_receiver(struct ivsh_dev_context *p_ivsh_dev_ctx, struct ivsh_ctrl_context *p_ivsh_ctrl_ctx)
+        {
+            struct ivsh_test_rx_context rx_ctx;
+            volatile struct pay_load_header *p_hdr;
+
+            /* Initialize the receiver related data */
+            ivsh_test_rx_init(&rx_ctx, p_ivsh_dev_ctx, p_ivsh_ctrl_ctx);
+            p_hdr = rx_ctx.p_hdr;
+            while (!is_p0_reset(p_hdr))
+                usleep(10000);
+            set_p1_status(p_hdr, SHMEM_STATUS_RESET);
+
+            while (!is_p0_initialized(p_hdr))
+                usleep(100);
+            set_p1_status(p_hdr, SHMEM_STATUS_INIT);
+            set_p0_status(p_hdr, SHMEM_STATUS_READY);
+
+            /* waiting for p0 write done */
+            ivsh_wait_irq(p_ivsh_dev_ctx, IVSH_TEST_VECTOR_ID);
+            ivsh_test_rx_recv(&rx_ctx);
+            usleep(100);
+
+            ivsh_test_rx_deinit(&rx_ctx);
+        }
+
+**The reference Sender and Receiver Sample Code based Polling mode:**
+
+     .. code-block:: none
```

Comment:
Same here.  Remove the code block indentation.

Reply:
Thanks, has been fixed.